### DISCUSSION TITLE
New Perl binding - Text::Treesitter

### DISCUSSION THREAD

**Discussion by leonerd** [tree-sitter/tree-sitter#2022] [Bindings]
Hello all,

I thought I'd announce I've started creating a Perl binding for tree-sitter. It can be seen at

https://metacpan.org/pod/Text::Treesitter

It currently wraps most of the actual `tree-sitter.h` API in some lower-level objects. I'm gradually working on putting a more convenient higher-level frontend API onto it, so it does things like remembering the original source string and each node can retrieve the substr of it that it relates to.

Additionally I'm also working on a `tree-sitter-perl` for actually attempting to parse Perl code - I'll announce that in a separate message.

**Comment by ahlinc** [comment]
@leonerd could you share a link to the binding source code and make a PR to the Tree-Sitter docs? The Tree-Sitter is an open source project and it isn't good to have something unclear in its docs.

**Reply by leonerd** [reply]
Do you mean this doc? https://github.com/tree-sitter/tree-sitter/blob/master/docs/index.md I see it already listed there:

```markdown
* [Perl](https://metacpan.org/pod/Text::Treesitter)
```

**Reply by ahlinc** [reply]
Yes, but Cpan isn't a source code repo, could you tell where is a source code repo?

**Reply by leonerd** [reply]
It's not the *upstream* canonical source, no. But it does have a "View Source" button that folks can read the source of that particular release. It's also the expected way that Perl users would interact with it. Installing from metacpan is verymuch "the usual process", whereas installing from the author's upstream source is not.
